gulp/tasks.js

```diff
@@ -35,7 +37,14 @@ gulp.task('watch', function (done) {
                'watch-assets', done)
 })
 
-gulp.task('test', function (done) {
-  gutil.log('Test that the app runs')
-  done()
+gulp.task('test', function () {
+  runSequence('generate-assets',
+              'mocha')
+})
+
+gulp.task('mocha', function () {
+  return gulp.src(['test/**/*.js'], { read: false })
+        .pipe(mocha({ reporter: 'spec' }))
+        .on('error', gutil.log)
```

Comment:
I think you can use:

```
  .pipe(mocha({reporter: 'spec'}))
  .once('error', () => {
    process.exit(1)
  })
  .once('end', () => {
    process.exit()
  })
```

https://github.com/sindresorhus/gulp-mocha#test-suite-not-exiting